File: app/src/main/java/com/expansion/lg/kimaru/expansion/other/InternetCheck.java
package com.expansion.lg.kimaru.expansion.other;


import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import static android.content.Context.CONNECTIVITY_SERVICE;

/**
 * Created by kimaru on 9/4/17.
 */

public class InternetCheck {
    Context context;

    public InternetCheck(Context context){
        this.context = context;
    }

    public boolean isConnected() {

        ConnectivityManager cm = (ConnectivityManager) context
                .getSystemService(Context.CONNECTIVITY_SERVICE);


        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
        return activeNetwork != null
                && activeNetwork.isConnectedOrConnecting();
    }
}

File: app/src/main/java/com/expansion/lg/kimaru/expansion/other/InternetCheck.kt
package com.expansion.lg.kimaru.expansion.other


import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.net.ConnectivityManager
import android.net.NetworkInfo

import android.content.Context.CONNECTIVITY_SERVICE

/**
 * Created by kimaru on 9/4/17.
 */

class InternetCheck(internal var context: Context) {

    val isConnected: Boolean
        get() {

            val cm = context
                    .getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager


            val activeNetwork = cm.activeNetworkInfo
            return activeNetwork != null && activeNetwork.isConnectedOrConnecting
        }
}